### DISCUSSION TITLE
Should i be concerned about these warnings

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#15731] [General]
Seems like we are accessing obsolete android sdk items?

`warning XAOBS001: 'ComponentActivity' is obsolete: 'While this type is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'ComponentActivity' is obsolete: 'While this type is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'ComponentActivity' is obsolete: 'While this type is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportBackgroundTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportBackgroundTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportBackgroundTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportBackgroundTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportCompoundDrawablesTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportCompoundDrawablesTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportCompoundDrawablesTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatButton.SupportCompoundDrawablesTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportBackgroundTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportBackgroundTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportButtonTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportButtonTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportButtonTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportButtonTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportCompoundDrawablesTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportCompoundDrawablesTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportCompoundDrawablesTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatCheckBox.SupportCompoundDrawablesTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'Fragment.MPreviousWho' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'Fragment.MPreviousWho' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'Fragment.HasOptionsMenu.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'Fragment.IsMenuVisible.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportBackgroundTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportBackgroundTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportBackgroundTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportBackgroundTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportButtonTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportButtonTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportButtonTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportButtonTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportCompoundDrawablesTintList.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportCompoundDrawablesTintList.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportCompoundDrawablesTintMode.get' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'
warning XAOBS001: 'AppCompatRadioButton.SupportCompoundDrawablesTintMode.set' is obsolete: 'While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.'`

**Comment by jeromelaban** [comment]
Thanks for the report. Those a recent warnings added by .NET that we're handling with https://github.com/unoplatform/uno/pull/15711, and they're not a concern at this point.

**Reply by cconner100** [reply]
OK anyway to make these quite so i dont get so much noise in the build?

**Reply by jeromelaban** [reply]
You can ignore these warnings by using:

```xml
<PropertyGroup>
   <NoWarn>$(NoWarn);XAOBS001</NoWarn>
</PropertyGroup>
```

**Reply by datvm** [reply]
@jeromelaban is that PR merged yet? I still have it now for `androidx.credentials.exceptions.GetCredentialException.Type` but I think it's fair because somehow there is no doc about that member at all https://developer.android.com/reference/kotlin/androidx/credentials/exceptions/GetCredentialException

**Reply by jeromelaban** [reply]
@datvm this PR was merged yes, but you may be getting this error at another location. If you can provide builds, we may find the location.

**Reply by datvm** [reply]
@jeromelaban hi sorry I just noticed this is Uno repo, not .NET/MAUI. I am using MAUI. I get the warning at `androidx.credentials.exceptions.GetCredentialException.Type`. Decompiled source:

```cs
		public virtual unsafe string Type {
			// Metadata.xml XPath method reference: path="/api/package[@name='androidx.credentials.exceptions']/class[@name='GetCredentialException']/method[@name='getType' and count(parameter)=0]"
			[global::System.Obsolete ("While this member is 'public', Google considers it internal API and reserves the right to modify or delete it in the future. Use at your own risk.", DiagnosticId = "XAOBS001")]
			[Register ("getType", "()Ljava/lang/String;", "GetGetTypeHandler")]
			get {
				const string __id = "getType.()Ljava/lang/String;";
				try {
					var __rm = _members.InstanceMethods.InvokeVirtualObjectMethod (__id, this, null);
					return JNIEnv.GetString (__rm.Handle, JniHandleOwnership.TransferLocalRef)!;
				} finally {
				}
			}
		}
```

**Reply by jeromelaban** [reply]
I see, then this is not something we can change, notifying the MAUI team would help.

**Reply by datvm** [reply]
@jeromelaban thanks!
